scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -14,29 +14,32 @@
 
 class HttpProxyMiddleware(object):
 
-    def __init__(self, auth_encoding='latin-1'):
+    def __init__(self, enabled=True, auth_encoding='latin-1'):
+        if not enabled:
+            raise NotConfigured
         self.auth_encoding = auth_encoding
         self.proxies = {}
         for type, url in getproxies().items():
             self.proxies[type] = self._get_proxy(url, type)
 
-        if not self.proxies:
-            raise NotConfigured
-
     @classmethod
     def from_crawler(cls, crawler):
+        enabled = crawler.settings.getbool('HTTPPROXY_ENABLED')
```

Comment:
It's probably simpler to just `raise NotConfigured` here directly, similarly to [how `CookiesMiddleware` does](https://github.com/scrapy/scrapy/blob/129421c7e31b89b9b0f9c5f7d8ae59e47df36091/scrapy/downloadermiddlewares/cookies.py#L23).